ui/src/tool-panel/location-track/location-track-infobox-duplicate-of.tsx

```diff
@@ -75,63 +142,99 @@ export const LocationTrackInfoboxDuplicateOf: React.FC<LocationTrackInfoboxDupli
     const { t } = useTranslation();
     const trackNumbers = useTrackNumbers(layoutContext);
+    const explicitDuplicateLocationTrackNames = useLocationTracks(
+        duplicatesOfLocationTrack
+            ?.map((d) => d.duplicateStatus.duplicateOfId)
+            ?.filter(filterNotEmpty) ?? [],
+        layoutContext,
+    );
+    const duplicateTrackNumberWarning =
+        existingDuplicate && currentTrackNumberId !== existingDuplicate?.trackNumberId
+            ? t('tool-panel.location-track.duplicate-on-different-track-number', {
+                  trackNumber: getTrackNumberName(trackNumbers, existingDuplicate.trackNumberId),
+              })
+            : '';
+
     return existingDuplicate ? (
-        <React.Fragment>
+        <span title={duplicateTrackNumberWarning}>
             <LocationTrackLink
                 locationTrackId={existingDuplicate.id}
                 locationTrackName={existingDuplicate.name}
             />
             &nbsp;
             {currentTrackNumberId !== existingDuplicate.trackNumberId && (
-                <LocationTrackDuplicateTrackNumberWarning
-                    differingTrackNumberName={getTrackNumberName(
-                        trackNumbers,
-                        existingDuplicate.trackNumberId,
-                    )}
-                />
+                <LocationTrackDuplicateInfoIcon type={'ERROR'} />
             )}
-        </React.Fragment>
+        </span>
     ) : duplicatesOfLocationTrack ? (
         <ul className={styles['location-track-infobox-duplicate-of__ul']}>
-            {duplicatesOfLocationTrack.filter(filterUniqueById((d) => d.id)).map((duplicate) => (
-                <li key={duplicate.id}>
-                    <LocationTrackLink
-                        locationTrackId={duplicate.id}
-                        locationTrackName={duplicate.name}
-                    />
-                    <span className={styles['location-track-infobox-duplicate-of__info-icons']}>
-                        {duplicate.duplicateStatus.duplicateOfId == undefined && (
-                            <LocationTrackDuplicateInfoIcon
-                                msg={t('tool-panel.location-track.implicit-duplicate-tooltip', {
-                                    trackName: duplicate.name,
-                                    otherTrackName: targetLocationTrack.name,
-                                })}
-                                type={'INFO'}
-                            />
-                        )}
-                        {currentTrackNumberId !== duplicate.trackNumberId && (
-                            <LocationTrackDuplicateTrackNumberWarning
-                                differingTrackNumberName={getTrackNumberName(
-                                    trackNumbers,
-                                    duplicate.trackNumberId,
-                                )}
+            {duplicatesOfLocationTrack.filter(filterUniqueById((d) => d.id)).map((duplicate) => {
```

Comment:
Koska duplikaattiraiteilla on voi olla useampi huomio, niin pistin tämän näyttämislogiikan aika lailla kokonaan uusiksi. Tämän myötä näytetään tosiaan vain max. yksi ikoni (virhe jos löytyy virheitä, info jos infoja.) Tooltipillä hoveroidessa näytetään tarkempi listaus, jossa huomiot on groupattu tyypeittäin ja näytetään ranskalaisilla viivoilla. Tuloksena on jotain seuraavan näköistä:

```
Raide "Choo Choo 001"
Virheet (2):
- Raide on eri ratanumerolla (1337)
- Joku toinen virhe

Huomiot (1):
- Joku huomio
```

Reply:
🤔 Tästä .map-kutsusta voisi varmaan jotain funktioida/laskea muualla (koodia aika reippaasti html-leiskan sisällä)